integrationtests/self/client_test.go

```diff
@@ -1,4 +1,4 @@
-package self_test
```

Comment:
Why are you dropping the _test suffix? I think go tools (mod, vendor) treat test packages differently.

Now that I think about it, maybe missing _test suffixes were the problem with the vendoring of test-only files?

Reply:
Then we probably should rename everything to `self_test` instead of `self`.